jax/tree_util.py

```diff
@@ -87,6 +87,25 @@ def treedef_children(treedef):
 def treedef_is_leaf(treedef):
   return treedef.num_nodes == 1
 
+def all_leaves(iterable):
```

Comment:
Does it make sense to add a `is_leaf` (in Python only)? I assume you preferred to add `AllLeaves` in C++ to that you don't have to come back to Python for every element of the iterable.

Reply:
Indeed I would prefer not to roundtrip through the interpreter if possible. I'm happy to add the function (as well as `all_leaves`) if you would like it.. Perhaps then a suitable name would be `all_is_leaf` since this is just a faster version of `all(map(is_leaf, x))`.

Reply:
I like `all_is_leaf`